import-automation/executor/commands/diff_imports.py

```diff
@@ -0,0 +1,158 @@
+                    default="datcom-204919",
+                    help=('GCP project id for base_bucket'))
+
+flags.DEFINE_string(name='head_bucket',
+                    default="datcom-dev-imports",
+                    help=('GCS bucket name of test data'))
+
+flags.DEFINE_string(name='head_bucket_project_id',
+                    default="google.com:datcom-store-dev",
+                    help=('GCP project id for head_bucket'))
+
+
+def _get_file_hashes(abs_import_name, bucket_project_id,
+                     bucket_name: str) -> Dict[str, str]:
+    """Returns a mapping of filepaths to md5 hash of the contents of latest folder."""
+    parts = abs_import_name.split(':')
+    if len(parts) != 2:
+        raise Exception('Invalid absolute import name %s' % abs_import_name)
+    path_from_script = '/'.join(parts)
+
+    client = storage.Client(project=bucket_project_id)
+    bucket = client.bucket(bucket_name)
+    version_file = f'{path_from_script}/latest_version.txt'
+    latest_version = bucket.blob(version_file).download_as_string().decode()
+
+    file_hashes = {}
+    for blob in client.list_blobs(
+            bucket_name, prefix=f'{path_from_script}/{latest_version}'):
+        # <path>/<version>/<file> -> <path>/<file>
+        canonical_name = blob.name.replace(latest_version + '/', '')
+        file_hashes[canonical_name] = blob.md5_hash
+    return file_hashes
+
+
+def _diff_file_hashes(abs_import_name, base_project_id, base_bucket,
+                      head_project_id, head_bucket: str):
+    """Returns error if there is diff of import between base_bucket and head_bucket."""
+    base_file_hashes = _get_file_hashes(abs_import_name, base_project_id,
+                                        base_bucket)
+    head_file_hashes = _get_file_hashes(abs_import_name, head_project_id,
+                                        head_bucket)
+
+    has_diff, missing, unexpected = [], [], []
+
+    for name, hash in base_file_hashes.items():
+        if name not in head_file_hashes:
+            missing.append(name)
+        elif hash != head_file_hashes[name]:
+            has_diff.append(name)
+
+    for name in head_file_hashes:
+        if name not in base_file_hashes:
+            unexpected.append(name)
+
+    return has_diff, missing, unexpected
+
+
+def main(argv):
+    for abs_import_name in FLAGS.abs_import_names:
+        trigger_job(
+            'google.com:datcom-store-dev',
+            'us-central1',
+            abs_import_name,
+            f'_{FLAGS.executor_type}'  # job id suffix
+        )
+
+        trigger_job(
+            'google.com:datcom-data',
+            'us-central1',
+            abs_import_name,
+            ''  # Legacy GAE has no job suffix
+        )
+
+    # There is no way to know when a job triggered by Cloud Scheduler finishes.
+    # Change
+    # Since the timeout is 30min, we sleep for 30min + some buffer time to wait
+    # for all jobs to finish.
```

Comment:
The code says 10 mins?

Reply:
Oops that was for testing small datasets, fixed and added more comments.